[CI] EC2RetriesTests testEC2DiscoveryRetriesOnRateLimiting failing

**Build Scans:**
- [elasticsearch-periodic #4293 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/yvevjkjy3hth4)
- [elasticsearch-periodic #4289 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/3trq3mp6icjse)
- [elasticsearch-periodic #4265 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/hp2rwcczcb2to)
- [elasticsearch-periodic #4253 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/aewh4sbl76imq)

**Reproduction Line:**
```
./gradlew ':plugins:discovery-ec2:test' --tests "org.elasticsearch.discovery.ec2.EC2RetriesTests.testEC2DiscoveryRetriesOnRateLimiting" -Dtests.seed=67E5E30A1DC80CD3 -Dtests.locale=en-TT -Dtests.timezone=Asia/Ashgabat -Druntime.java=23
```

**Applicable branches:**
7.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testEC2DiscoveryRetriesOnRateLimiting'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.discovery.ec2.EC2RetriesTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a collection with size <1>
     but: collection size was <0>
```

**Issue Reasons:**
- [7.17] 4 consecutive failures in step openjdk23_java-matrix
- [7.17] 4 failures in test testEC2DiscoveryRetriesOnRateLimiting (1.0% fail rate in 416 executions)
- [7.17] 4 failures in step openjdk23_java-matrix (100.0% fail rate in 4 executions)
- [7.17] 4 failures in pipeline elasticsearch-periodic (36.4% fail rate in 11 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/EC2RetriesTests.java
+++ b/plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/EC2RetriesTests.java
@@ -101,6 +101,7 @@ public class EC2RetriesTests extends AbstractEC2MockAPITestCase {
                     exchange.getResponseHeaders().set("Content-Type", "text/xml; charset=UTF-8");
                     exchange.sendResponseHeaders(HttpStatus.SC_OK, responseBody.length);
                     exchange.getResponseBody().write(responseBody);
+                    exchange.getResponseBody().flush();
                     return;
                 }
             }

